ctapipe/image/image_processor.py

```diff
@@ -169,8 +169,9 @@ def _process_telescope_event(self, event):
         """
         Loop over telescopes and process the calibrated images into parameters
         """
-
         for tel_id, dl1_camera in event.dl1.tel.items():
+            geometry = self.subarray.tel[tel_id].camera.geometry
+            geometry_telescope = geometry.transform_to(self.telescope_frame)
```

Comment:
This is a pretty slow operation (transforming all pixels), and since nothing really changes per event, it could be pre-computed once. Only the focal length matters here, not the pointing, as it is still relative to the camera center.   So we could either store the TelescopeFrame Geometries in a cache in the ImageProcessor (once per telescope).